### PR TITLE
cross-platform: Fix GCC lib Fedora X64 compile time issues

### PR DESCRIPTION
**Pull request by obastemur**
Fedora x64, we have only 1 failing test at the moment: 'test/Regex/BoiHardFail.js'

Compile steps for Fedora 24

Install dependencies
```
sudo dnf group install -y "Development Tools" "C Development Tools and Libraries"
sudo dnf install -y git cmake clang gcc gcc-c++ kernel-devel python
sudo dnf install -y libuuid-devel lttng-ust-devel.x86_64  libunwind-devel.x86_64 libicu-devel.x86_64
```

Clone && Compile
```
> git clone https://github.com/Microsoft/ChakraCore
> cd ChakraCore
> ./build.sh
```

### CODE REVIEW

File: pal/src/include/pal/palinternal.h [hunk 1]
```diff
@@ -677,4 +677,11 @@ inline T* InterlockedCompareExchangePointerT(
 
 #endif // __cplusplus
 
+#ifndef max
```
Comment (digitalinfinity) [inline pal/src/include/pal/palinternal.h]: Curious, why is this needed for fedora and not needed for ubuntu? pal.h also defines min and max, so this seems unnecessary but I saw the same compile error as you. Can you explain why that definition is insufficient?

**Comment by digitalinfinity** [comment]
Awesome. Thanks for fixing this @obastemur! LGTM, just had one question.

File: pal/src/include/pal/palinternal.h (hunk 1 above)
Comment (obastemur) [inline pal/src/include/pal/palinternal.h]: IIRC; NO_PAL_MINMAX was being defined in the chain by mistake.

**Comment by obastemur** [comment]
@digitalinfinity @jianchun thanks for the reviews!
